webapp/channels/src/components/autocomplete_selector.tsx

```diff
@@ -1,55 +1,68 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import SuggestionBox from 'components/suggestion/suggestion_box';
 import SuggestionList from 'components/suggestion/suggestion_list';
 
-export default class AutocompleteSelector extends React.PureComponent {
-    static propTypes = {
-        providers: PropTypes.array.isRequired,
-        value: PropTypes.string.isRequired,
-        onSelected: PropTypes.func,
-        label: PropTypes.node,
-        labelClassName: PropTypes.string,
-        inputClassName: PropTypes.string,
-        helpText: PropTypes.node,
-        placeholder: PropTypes.string,
-        footer: PropTypes.node,
-        disabled: PropTypes.bool,
-        toggleFocus: PropTypes.func,
-        listComponent: PropTypes.elementType,
-        listPosition: PropTypes.string,
-    };
+import type ModalSuggestionList from './suggestion/modal_suggestion_list';
+import type Provider from './suggestion/provider';
+
+type Props = {
+    id?: string;
+    providers: Provider[];
+    value: string;
+    onSelected?: (selected: Selected) => void;
+    label?: React.ReactNode | string;
+    labelClassName?: string;
+    inputClassName?: string;
+    helpText?: React.ReactNode | string;
+    placeholder?: string;
+    footer?: Node;
+    disabled?: boolean;
+    toggleFocus?: ((focus: boolean) => void) | null;
+    listComponent?: typeof SuggestionList | typeof ModalSuggestionList;
+    listPosition?: string;
+};
+
+type State = {
+    input: string;
+    focused?: boolean;
+};
+
+type Selected = {
```

Comment:
It is a bit out of the scope, but I have mixed feelings with this.

"Selected" is defined all over the place.

In `action_menu` is defined as `Option | UserProfile | Channel`
In `select_binding` is defined as `Option`
In `dialog_element` is defined as the way it is defined here.

We should centralize this type somehow (probably in this file, since it should be used only when the autocomplete selector is imported), and use always the same.

I feel the correct type in this case is `Option | UserProfile | Channel`, but 0/5.

Thoughts @M-ZubairAhmed ?

Reply:
I also believe the correct type is `Option | UserProfile | Channel`. I also used it at first, but it required changes in the files you mentioned, so I went with this solution which didn't require changes.

Reply:
I have the change ready, I can push it if needed.

Reply:
If the change doesn't cause other typing issues, I'd be good with that. Ideally, I think this component would have some parameterized type like `AutocompleteSelector<UserProfile>`, but React seems to have trouble with those, so using an option type to avoid repeating ourselves sounds good to me